packages/itwinui-css/src/dialog/dialog.scss

```
@@ -5,14 +5,18 @@
 $iui-dialog-width: 380px;
 
 @mixin iui-dialog-wrapper {
-  position: relative;
+  position: fixed;
   overflow: hidden;
   top: 0;
   left: 0;
   width: 100%;
   height: 100%;
   pointer-events: none;
   transform: translateX(0); // creates containing block for position: fixed
+
+  &[data-iui-relative] {
```

How about targeting when equal to `true`?
```suggestion
  &[data-iui-relative='true'] {
```